Ports/.port_include.sh

```diff
@@ -366,6 +370,38 @@ fetch_simple() {
     fi
 }
 
+fetch_git() {
+    repository="${1}"
+    revision="${2}"
+
+    directory="$(basename "${repository}")"
+    backing_copy="${PORT_META_DIR}/${directory}"
+    working_copy="${PORT_BUILD_DIR}/${directory}"
+
+    run_nocd git init --bare "${backing_copy}"
+    run_nocd git -C "${backing_copy}" config core.autocrlf false
+    run_nocd git -C "${backing_copy}" worktree prune
+    run_nocd git -C "${backing_copy}" fetch --tags "${repository}" "${revision}"
```

Comment:
For a standard (non-dev) checkout of a git repository, we can probably supply `--depth=1` here on initial fetch to minimize bandwidth and storage?

Reply:
We could, but not all ports (currently 0%; sample size 1) play nicely with that. For example, `gn` expects that, if you are building from inside Git, all the history down to the `initial-commit` tag is available.

At least `git fetch --unshallow` is a thing now, so if we decide to do it, it's going to be relatively painless.

Reply:
Let's not then :^)